### PR TITLE
Istanbul eip 1108 gas costs

### PR DESCRIPTION
**Pull request by Shadowfiend**
@pdyraga @ngrinkevich mind checking for anything egregious here?

See keep-network/keep-core#963.

### CODE REVIEW

File: params/protocol_params.go [hunk 1]
```diff
@@ -73,18 +73,25 @@ const (
...
+	Sha256BaseGas           uint64 = 60    // Base price for a SHA256 operation
+	Sha256PerWordGas        uint64 = 12    // Per-word price for a SHA256 operation
+	Ripemd160BaseGas        uint64 = 600   // Base price for a RIPEMD160 operation
+	Ripemd160PerWordGas     uint64 = 120   // Per-word price for a RIPEMD160 operation
+	IdentityBaseGas         uint64 = 15    // Base price for a data copy operation
+	IdentityPerWordGas      uint64 = 3     // Per-work price for a data copy operation
+	ModExpQuadCoeffDiv      uint64 = 20    // Divisor for the quadratic particle of the big int modular exponentiation
+	Bn256AddGas             uint64 = 150   // Gas needed for an elliptic curve addition
+	Bn256ScalarMulGas       uint64 = 6000  // Gas needed for an elliptic curve scalar
+	Bn256PairingBaseGas     uint64 = 45000 // Base price for an elliptic curve pairing
```
Comment (pdyraga) [inline params/protocol_params.go]: ... curve pairing *check*?

File: params/protocol_params.go [hunk 2]
```diff
@@ -73,18 +73,25 @@ const (
...
+	EcrecoverGas            uint64 = 3000  // Elliptic curve sender recovery gas price
+	Sha256BaseGas           uint64 = 60    // Base price for a SHA256 operation
+	Sha256PerWordGas        uint64 = 12    // Per-word price for a SHA256 operation
+	Ripemd160BaseGas        uint64 = 600   // Base price for a RIPEMD160 operation
+	Ripemd160PerWordGas     uint64 = 120   // Per-word price for a RIPEMD160 operation
+	IdentityBaseGas         uint64 = 15    // Base price for a data copy operation
+	IdentityPerWordGas      uint64 = 3     // Per-work price for a data copy operation
+	ModExpQuadCoeffDiv      uint64 = 20    // Divisor for the quadratic particle of the big int modular exponentiation
+	Bn256AddGas             uint64 = 150   // Gas needed for an elliptic curve addition
+	Bn256ScalarMulGas       uint64 = 6000  // Gas needed for an elliptic curve scalar
```
Comment (pdyraga) [inline params/protocol_params.go]: ... scalar *multiplication*?

**Comment by pdyraga** [comment]
I think you may need to alter this piece:
https://github.com/keep-network/go-ethereum/blob/master/eth/tracers/tracer.go#L393
as well.

**Comment by pdyraga** [comment]
Three comments, the rest looks good.

**Comment by ngrinkevich** [comment]
Getting flashbacks from 2018 😄On a serious note the code looks great!

**Comment by Shadowfiend** [comment]
> I think you may need to alter this piece:

Not sure because I'm not sure the tracer should be behaving as Istanbul right now vs closer to the hardfork. Will see what the go-ethereum folks think on this one.

**Comment by Shadowfiend** [comment]
Thanks for the catch on those two typos, fixed, closing here, and opening on go-ethereum.

**Pull request closed without merging**